src/WooCommerce.php

```diff
@@ -128,29 +128,35 @@ public static function woocommerce_get_availability($stock, $product) {
     $product->set_manage_stock('yes');
     $product->set_backorders('yes');
 
-    // Low stock level is only set on parent product.
-    // If product is a variation then manually get the value based on parent ID.
+    // If low stock threshold is not set at product level, get global option.
     if ($product->is_type('variation')) {
-      $low_stock_amount = get_post_meta($product->get_parent_id(), '_low_stock_amount', TRUE);
+      $parent_product = wc_get_product($product->get_parent_id());
+      $low_stock_amount = $parent_product->get_low_stock_amount();
     }
     else {
       $low_stock_amount = $product->get_low_stock_amount();
     }
+    if ($low_stock_amount === '') {
```

Comment:
Can `$low_stock_amount` be undefined at this point? If yes, we should check it with `isset`.

Also, why not using `empty` PHP method? AFAIK, the only difference is that `empty` will consider a string containing `0` as `TRUE`. Your solution is instead checking only an exactly empty string.

If we should consider `0` as a falsy value, I think we should make it more explicit in the code (by casting) or adding a comment.

Reply:
I actually copied these lines from WooCommerce Core - https://github.com/woocommerce/woocommerce/blob/master/includes/wc-stock-functions.php#L340-L343

I thought it was better to stick with the Core method but I guess that `empty()` would be better. I'll update 👍

Reply:
Fixed.

Reply:
As we discussed in person, we need to not consider `0` value for that conditional.
Therefore, @colourgarden is going to revert the fix and explain in a comment why we are only checking for empty strings.

Reply:
After discussing offline further, we decided to revert to the WooCommerce empty string check as zero is actually a valid value for the WooCommerce field.